sycl/source/detail/sycl_mem_obj_t.cpp

```diff
@@ -90,6 +90,15 @@ void SYCLMemObjT::updateHostMemory() {
         pi::cast<RT::PiMem>(MInteropMemObject));
   }
 }
+
+// TODO: With ABI break the attached resources can be held by this type. When
+// that happens this will be obsolete as the resources will automatically be
+// destroyed with the object.
+void SYCLMemObjT::detachResources() const {
+  if (MInteropContext)
```

Comment:
Could you please tell why this check is required?

Reply:
It does not appear to be guaranteed that all paths are guaranteed to have a context associated, but if it is not there then we know there will be no attached resources. It is likely not applicable to buffer and image, but to make destruction as simple as possible I think it makes the most sense to do the check here.

Reply:
Above statement is wrong and has been addressed in latest changes.